ReactSkia/views/common/RSkTextUtils.cpp

```diff
@@ -77,27 +78,38 @@ void drawText(std::shared_ptr<Paragraph>& paragraph,
             const ParagraphProps& props,
             bool isParent) {
     SkPaint paint;
+    SkPath clipPath;
     SkScalar yOffset = 0;
     Rect frame = layout.frame;
-    auto fontLineHeight = (!std::isnan(props.textAttributes.lineHeight)) && (props.textAttributes.lineHeight >= 0) ?
-                            props.textAttributes.lineHeight :
-                            true;
+    int textFrameHeight=(std::isnan(props.textAttributes.lineHeight)) ? frame.size.height : props.textAttributes.lineHeight;
+    SkAutoCanvasRestore save(canvas, true);
+    
+    if(textFrameHeight <= 0){
+        RNS_LOG_WARN(" Text Drawing Ignored, Height[" << textFrameHeight<<"]");
+        return;
+    }
 
-    if ((props.backgroundColor) || (props.textAttributes.lineHeight >= 0)) {
-        if (isParent)
-            canvas->clipRect(SkRect::MakeXYWH(0, 0, frame.size.width, frame.size.height));
-        else
-            canvas->clipRect(SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height));
-        canvas->drawColor(RSkColorFromSharedColor(props.backgroundColor, SK_ColorTRANSPARENT));
+    if (isParent){
+        clipPath.addRect(0, 0, frame.size.width, textFrameHeight);
+    } else {
+        clipPath.addRect(frame.origin.x, frame.origin.y, frame.origin.x + frame.size.width, frame.origin.y + textFrameHeight);
     }
-    if (fontLineHeight) {
+    canvas->clipPath(clipPath, SkClipOp::kIntersect);
```

Comment:
Why clipath is used instead of clipRect, As there is only one Rect. ANy specific Reason ??

Reply:
Good catch. Yes we can use clipRect here with kIntersect